src/nat/middleware/cache/cache_middleware.py

```
@@ -84,20 +82,33 @@ def __init__(self, *, enabled_mode: str, similarity_threshold: float) -> None:
+        """Check if caching should be enabled based on configuration and context."""
         if self._enabled_mode == "always":
             return True
 
-        # Get the current context and check if we're in evaluation mode
+        # For "eval" mode, only enable when in evaluation context
         try:
             context_state = ContextState.get()
-            context = Context(context_state)
-            return context.is_evaluating
+            ctx = Context(context_state)
+            return ctx.is_evaluating
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)